framework/Samplers/CustomSampler.py

```diff
@@ -53,6 +53,8 @@ class cls.
     sourceInput.addParam("class", InputData.StringType)
     inputSpecification.addSub(sourceInput)
 
+    inputSpecification.addSub(InputData.parameterInputFactory('index', contentType=InputData.IntegerListType))
```

Comment:
It is a IntegerListType now, so no need to split for ```,``` at line 107.

Reply:
That would be true, except it appears to still be reading off the raw XML; it hasn't been converted yet, and I didn't have the time to do it.

Reply:
```
class IntegerListType(InputType):
  """
    A type for integer lists "1, 2, 3" -> [1,2,3]
  """

  @classmethod
  def convert(cls, value):
    """
      Converts value from string to an integer list.
      @ In, value, string, the value to convert
      @ Out, convert, list, the converted value
    """
    return [int(x.strip()) for x in value.split(",")]

#Note, XSD's list type is split by spaces, not commas, so using xsd:string
IntegerListType.createClass("stringtype","xsd:string")
```

I see the above.... what do you mean?

Reply:
As we talked about in person, while the CustomSampler implements the InputParams, it does not use them, since it overwrites the "readMoreXmlBase" method.